Fix appveyor

Zialus opened the pull request:
I'm back with CI fixes.

This should finally make everything build fine on windows.

ddemidov commented on .appveyor.yml:
```diff
@@ -1,40 +1,31 @@
-image:
-  - Visual Studio 2017
-  - Ubuntu
```
Can we keep Ubuntu in the matrix?

Zialus replied:
I was thinking you might be interested in leaving linux/macOS to travis and windows to appveyor. But I can add it back.

ddemidov replied:
I was going to say that someone added Ubuntu support just recently, so I would not want to waste their effort, but apparently that someone was [you](#261), so I'll leave this for you to decide :).

Zialus replied:
Actually there is still a use for it that Travis doesn't cover. I'll add it back after the VS2013 thing gets figured out.

ddemidov commented on .appveyor.yml:
```diff
@@ -1,40 +1,31 @@
+      GENERATOR: Visual Studio 14 2015 Win64
+      BOOST_ROOT: C:\Libraries\boost_1_63_0
+    - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2013
+      GENERATOR: Visual Studio 12 2013 Win64
```
It looks like there is a problem with VS2012 and boost.multi_index.

ddemidov commented:
Thank you! I've left a couple of notes in the code.

Zialus commented:
VS2013 just doesn't seem to work

https://ci.appveyor.com/project/Zialus/vexcl/builds/21238879/job/io39ts666vbrestm

I've added Ubuntu back and removed VS2013

ddemidov commented:
Thanks again!
